server/chat-plugins/scavenger-games.ts

```diff
@@ -438,6 +445,130 @@ const TWISTS: {[k: string]: Twist} = {
+
+			number--; // indexOf starts at 0
+
+			if (questionAnswer === 'answer') {
+				// These two lines are the only difference from the original
+				this.mines[number] = answer.filter(ans => ans.startsWith('!'));
+				this.questions[number].answer = answer.filter(ans => !ans.startsWith('!'));
+			} else {
+				this.questions[number].hint = value;
+			}
+
+			this.announce(`The ${questionAnswer} for question ${number + 1} has been edited.`);
+			if (questionAnswer === 'hint') {
+				for (const p in this.playerTable) {
+					this.playerTable[p].onNotifyChange(number);
+				}
+			}
+			return true;
+		},
+
+		onAnySubmit(player) {
+			if (this.huntLocked) {
+				player.sendRoom('The hunt was not set up correctly.  Please wait for the host to reset the hunt and create a new one.');
+				return true;
+			}
+		},
+
+		onIncorrectAnswer(player: ScavengerHuntPlayer, value: string) {
+			const curr = player.currentQuestion;
+
+			if (!this.guesses[curr][player.id]) this.guesses[curr][player.id] = new Set();
+			this.guesses[curr][player.id].add(toID(value));
+
+			throw new Chat.ErrorMessage("That is not the answer - try again!");
+		},
+
+		onShowEndBoard(endedBy?: User) {
+			const sliceIndex = this.gameType === 'official' ? 5 : 3;
+			const hosts = Chat.toListString(this.hosts.map(h => `<em>${Utils.escapeHTML(h.name)}</em>`));
+
+			const mines: {mine: string, users: string[]}[][] = [];
+
+			for (let index = 0; index < this.mines.length; index++) {
+				mines[index] = [];
+				for (const mine of this.mines[index]) {
+					mines[index].push({mine: mine.substr(1), users: []});
+				}
+			}
+
+			for (const player of Object.values(this.playerTable)) {
+				if (player.mines) {
+					for (const {index, mine} of player.mines) {
+						mines[index].find(obj => obj.mine === mine)?.users.push(player.name);
+					}
+				}
+			}
+
+			for (const mineSet of mines) mineSet.sort();
+
+			this.announce(
+				`The ${this.gameType ? `${this.gameType} ` : ""}scavenger hunt by ${hosts} was ended ${(endedBy ? "by " + Utils.escapeHTML(endedBy.name) : "automatically")}.<br />` +
+				`${this.completed.slice(0, sliceIndex).map((p, i) => `${Utils.formatOrder(i + 1)} place: <em>${Utils.escapeHTML(p.name)}</em> <span style="color: lightgreen;">[${p.time}]</span>.<br />`).join("")}${this.completed.length > sliceIndex ? `Consolation Prize: ${this.completed.slice(sliceIndex).map(e => `<em>${Utils.escapeHTML(e.name)}</em> <span style="color: lightgreen;">[${e.time}]</span>`).join(', ')}<br />` : ''}<br />` +
+				`<details style="cursor: pointer;"><summary>Solution: </summary><br />${this.questions.map((q, i) => `${i + 1}) ${Chat.formatText(q.hint)} <span style="color: lightgreen">[<em>${Utils.escapeHTML(q.answer.join(' / '))}</em>]</span><br/><details style="cursor: pointer;"><summary>Mines: </summary>${mines[i].map(({mine, users}) => Utils.escapeHTML(`${mine}: ${users.join(' / ') || '-'}`)).join('<br />')}</details>`).join("<br />")}</details>`
```

Comment:
Please break these up into more lines.

Reply:
Most of that is copy-pasted from the original code; should I modify both?

Reply:
That'd be ideal - it's fairly illegible to keep it all in one line.